rust/src/ssh/ssh.rs

```diff
@@ -149,6 +159,30 @@ impl SSHState {
         while !input.is_empty() {
             match parser::ssh_parse_record(input) {
                 Ok((rem, head)) => {
+                    let _pdu = Frame::new(
```

Comment:
I think it would make sense to move this outside of `Ok`, as we can want the frames in malformed data as well. I suppose you would just open all 3 (RecordHdr with fixed len, the other two with -1 len), then set the final length on `Ok`.

In general, I am missing an explanation of the frames in the commit message.

Also should add docs for the frames, cf https://docs.suricata.io/en/latest/rules/smtp-keywords.html#frames

Reply:
> I think it would make sense to move this outside of Ok, as we can want the frames in malformed data as well. I suppose you would just open all 3 (RecordHdr with fixed len, the other two with -1 len), then set the final length on Ok.

I am not sure it makes sense for SSH :
- malformed data can only be "negative length" for SSH record data (in SSH, there is the anti pattern that the length includes the fixed header length), and in this case, the flow is placed into error. So it is then impossible to "set the final length on Ok."
- for incomplete data, it is already handled in one way

Should I spend some time again to try to make a SV test that shows a different behavior with -1 len ?

> In general, I am missing an explanation of the frames in the commit message.

Adding something

> Also should add docs for the frames

I thought frames were automatically documented, by some rustdocs magic with `// app-layer-frame-documentation tag start: FrameType enum`
cc @jasonish ?

Add adding something like smtp

Reply:
See https://github.com/OISF/suricata/pull/11451/commits/b8462697a68388640bb558bc09f66333bdc391a1

I had interpreted this wrongly